infrastructure/gsuite-exporter/gsuite_exporter/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,143 @@
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import math
+import dateutil.parser
+import logging
+from gsuite_exporter import auth
+from gsuite_exporter.exporters.base import BaseExporter
+
+logger = logging.getLogger(__name__)
+
+class StackdriverExporter(BaseExporter):
+    """Convert Admin SDK logs to logging entries and send them to Stackdriver
+    Logging API.
+
+    Args:
+        api (`googleapiclient.discovery.Resource`): The Admin SDK API to fetch
+            records from.
+        version (str): The Admin SDK API version.
+        credentials_path (str, optional): The path to the GSuite Admin credentials.
+    """
+    SCOPES = [
+        'https://www.googleapis.com/auth/logging.read',
+        'https://www.googleapis.com/auth/logging.write'
+    ]
+    LOGGING_API_VERSION = 'v2'
+    def __init__(self,
+                 project_id,
+                 credentials_path=None):
+        logger.debug("Initializing Stackdriver Logging API ...")
+        self.api = auth.build_service(
+            api='logging',
+            version=StackdriverExporter.LOGGING_API_VERSION,
+            credentials_path=credentials_path,
+            scopes=StackdriverExporter.SCOPES)
+        self.project_id = "projects/{}".format(project_id)
+
+    def send(self, records, log_name, dry=False):
+        """Writes a list of Admin SDK records to Stackdriver Logging API.
+
+        Args:
+            records (list): A list of log records.
+            log_name (str): The log name to write (e.g: 'logins').
+            dry (bool): Toggle dry-run mode (default: False).
+
+        Returns:
+            `googleapiclient.http.HttpRequest`: The API response object.
+        """
+        res = None
+        destination = self.get_destination(log_name)
+        if records:
+            entries = self.convert(records)
+            body = {
+                'entries': entries,
+                'logName': '{}'.format(destination),
+                'dryRun': dry
+            }
+            logger.debug("Writing {} entries to Stackdriver Logging API @ '{}'".format(
+                len(entries),
+                destination))
+            res = self.api.entries().write(body=body).execute()
+        return res
+
+    def convert(self, records):
+        """Convert a bunch of Admin API records to Stackdriver Logging API
+        entries.
+
+        Args:
+            records (list): A list of Admin API records.
+
+        Returns:
+            list: A list of Stackdriver Logging API entries.
+        """
+        return map(lambda i: self.__convert(i), records)
```

Comment:
What version of Python is this meant to be run on?  In Py3 this will not return a list, it'll return an iterator.  I don't see a version in the README and the setup.py doesn't use `python_requires`.

Reply:
I've ran it on Python 2.7. Compatibility with Python 3 would be a good thing.
In the meantime, I've added `python_requires='>=2.7,<3.0'` to our `setup.py`